lib/web_ui/lib/src/engine/window.dart

```diff
@@ -708,6 +734,25 @@ class ViewConstraints implements ui.ViewConstraints {
       minHeight = size.height,
       maxHeight = size.height;
 
+  /// Converts JsViewConstraints into ViewConstraints.
+  ///
+  /// Since JsViewConstraints are expressed by the user, in logical pixels, this
+  /// conversion uses logical pixels for the available size as well. The resulting
+  /// ViewConstraints object can be multiplied by devicePixelRatio later to compute
+  /// the physicalViewConstraints.
+  factory ViewConstraints.fromJs(
+    JsViewConstraints? constraints, ui.Size availableLogicalSize) {
+    if (constraints == null) {
+      return ViewConstraints.tight(availableLogicalSize);
+    }
+    return ViewConstraints(
+      minWidth: _computeMinConstraintValue(constraints.minWidth, availableLogicalSize.width),
```

Comment:
Do we need to check that `constraints.minWidth <= availableLogicalSize.width` etc?

Reply:
I experimented with this, but we can't do it. Imagine that you want to render a flutter app in an empty div (0px tall) that has min-height: 20px. In that case the validation would fail, but it really shouldn't IMO. It should respect min-height 20px, and resize the container div to 20px tall.

Same with max-height, it must be allowed to be larger than the available size, so flutter can outgrow its container.

Reply:
Ah I get it now. In that case, I think the word `available` in the name is a bit misleading. I like `current` better in my personal opinion (similar to what it's called at the callsite in the `physicalConstraints` getter).

Reply:
Can rename, `current` definitely was in the code at some point :P

Reply:
Done!